### ISSUE TITLE
Should contentEditable elems be excempt from no-noninteractive-element-interactions?

### ISSUE THREAD

**Issue by svinkle** [jsx-eslint/eslint-plugin-jsx-a11y#525]
I have a `pre` element set as `contentEditable` for reasons. Thoughts on removing `contentEditable` elements from the [no-noninteractive-element-interactions](https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/no-noninteractive-element-interactions.md) rule?

I also added `role="textbox"` since that's really what it is, but I have other `input`s inserting content in specific areas of the `contentEditable` elem; this is why it's _not_ an actual `textarea`. However, doing so comes back with the [no-noninteractive-element-to-interactive-role](https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/no-noninteractive-element-to-interactive-role.md) error.

Here's the snippet in question:

```html
<pre
  aria-label="Edit description"
  aria-multiline="true"
  contentEditable
  onKeyUp={this.doSomething}
  role="textbox"
>
  <!-- 
    Other HTML which builds the content of this container
    which, in turn, can also be edited directly.
  -->
</pre>
```

My thoughts are since this has `contentEditable` it's now an interactive element and could be exempt from this rule.

**Comment by ari-os310** [comment]
@ljharb I've been looking through this issue and reading through the rules mentioned above, along with the [isNoninteractiveElement](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/f702f627a2afdbce0184b5bcad66be969ae8aa6e/src/util/isNonInteractiveElement.js#L143), and the [isInteractiveRole](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/f702f627a2afdbce0184b5bcad66be969ae8aa6e/src/util/isInteractiveRole.js). I'm thinking these are the files (or two of) the files I should be looking at to resolve this issue, but I'm not certain. Am I moving in the correct direction, and if not, where do you suggest I should be looking? Gracias !

**Comment by ljharb** [comment]
@ari-os310 that seems right to me, but tbh i'd just make the test cases to start, and then tinker til i got them passing :-)
